service-users-specs/src/test/java/io/pestakit/users/api/spec/steps/UsersSteps.java

```diff
@@ -0,0 +1,165 @@
+import cucumber.api.java.en.When;
+import io.pestakit.users.ApiException;
+import io.pestakit.users.ApiResponse;
+import io.pestakit.users.api.DefaultApi;
+import io.pestakit.users.api.dto.User;
+import io.pestakit.users.api.spec.helpers.Environment;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class UsersSteps {
+
+    protected Environment environment;
+    protected DefaultApi api;
+
+    User user;
+    List<User> users;
+
+    protected ApiResponse lastApiResponse;
+    protected ApiException lastApiException;
+    protected boolean lastApiCallThrewException;
+    protected int lastStatusCode;
+
+    public UsersSteps(Environment environment) {
+        this.environment = environment;
+        this.api = environment.getApi();
+    }
+
+    @Given("^there is a User and Team Server$")
+    public void thereIsAUserAndTeamServer() throws Throwable {
+        assertNotNull(api);
+    }
+
+    @When("^I GET the /users endpoint$")
+    public void iGETTheUsersEndpoint() throws Throwable {
+        try {
+            lastApiResponse = api.getUsersWithHttpInfo();
+            lastApiCallThrewException = false;
+            lastApiException = null;
+            lastStatusCode = lastApiResponse.getStatusCode();
+        } catch (ApiException e) {
+            lastApiCallThrewException = true;
+            lastApiResponse = null;
+            lastApiException = e;
+            lastStatusCode = lastApiException.getCode();
+        }
+    }
+
+    @Then("^I receive a (\\d+) status code$")
+    public void i_receive_a_status_code(int code) throws Throwable {
+        assertEquals(code, lastStatusCode);
+    }
+
+    @Given("^I have a user payload$")
+    public void iHaveAUserPayload() throws Throwable {
+        iHaveAUserNamedPayload("john");
+    }
+
+    @When("^I POST it to the /users endpoint$")
+    public void iPOSTItToTheUsersEndpoint() throws Throwable {
+        try {
+            lastApiResponse = api.createUserWithHttpInfo(user);
+            lastApiCallThrewException = false;
+            lastApiException = null;
+            lastStatusCode = lastApiResponse.getStatusCode();
+        } catch (ApiException e) {
+            lastApiCallThrewException = true;
+            lastApiResponse = null;
+            lastApiException = e;
+            lastStatusCode = lastApiException.getCode();
+        }
+    }
+
+    @Given("^I have a user named (.*?) payload$")
+    public void iHaveAUserNamedPayload(String name) throws Throwable {
+        if (name.indexOf('/') != -1) {
+            name = "ERROR";
+        }
+        user = new User();
+        user.username("user"+name);
+        user.password("pass"+name);
+        user.setEmail(name+"@"+name+".com");
+        user.setFirstName("first"+name);
+        user.setLastName("last"+name);
+        user.setDisplayName("display"+name);
+
+    }
+
+    @Then("^I receive an endpoint to the user$")
```

Comment:
"I receive an endpoint to the user"... what does it mean? Please find a better sentence.

Reply:
b36b4fc fixing this